azure-mgmt-batch/src/main/java/com/microsoft/azure/management/batch/BatchAccount.java

```diff
@@ -0,0 +1,205 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.management.batch;
+
+import com.microsoft.azure.CloudException;
+import com.microsoft.azure.management.batch.implementation.AccountResourceInner;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.GroupableResource;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.Resource;
+import com.microsoft.azure.management.resources.fluentcore.model.Creatable;
+import com.microsoft.azure.management.resources.fluentcore.model.Refreshable;
+import com.microsoft.azure.management.resources.fluentcore.model.Updatable;
+import com.microsoft.azure.management.resources.fluentcore.model.Wrapper;
+import com.microsoft.azure.management.resources.fluentcore.model.Appliable;
+import com.microsoft.azure.management.storage.StorageAccount;
+
+import java.io.IOException;
+
+/**
+ * An immutable client-side representation of an Azure batch account.
+ */
+public interface BatchAccount extends
+        GroupableResource,
+        Refreshable<BatchAccount>,
+        Updatable<BatchAccount.Update>,
+        Wrapper<AccountResourceInner> {
+
+    /**
+     * @return the provisioned state of the resource. Possible values include:
+     * 'Invalid', 'Creating', 'Deleting', 'Succeeded', 'Failed', 'Cancelled'
+     */
+    AccountProvisioningState provisioningState();
+
+    /**
+     * @return Get the accountEndpoint value.
+     */
+    String accountEndpoint();
+
+    /**
+     * @return the properties and status of any auto storage account associated with
+     * the account
+     */
+    AutoStorageProperties autoStorage();
+
+    /**
+     * @return the core quota for this BatchAccount account
+     */
+    int coreQuota();
+
+    /**
+     * @return the pool quota for this BatchAccount account
+     */
+    int poolQuota();
+
+    /**
+     * @return the active job and job schedule quota for this BatchAccount account
+     */
+    int activeJobAndJobScheduleQuota();
+
+    /**
+     * @return the access keys for this batch account
+     *
+     * @throws CloudException exception thrown from REST call
+     * @throws IOException exception thrown from serialization/deserialization
+     */
+    BatchAccountKeys keys() throws CloudException, IOException;
```

Comment:
Changed it to set of 2 functions keys and refreshKeys which are similar to Storage key apis.